dune-project

```diff
@@ -49,7 +49,7 @@
   (bindlib
    (>= 5.0.1))
   (cmdliner
-   (= 1.0.4))
+   (>= 1.1.0))
```

Comment:
So last time we tried that it broke the Nix build and we had to revert: https://github.com/CatalaLang/catala/pull/206. Tagging @lIlIlIlIIIIlIIIllIIlIllIIllIII for support on that issue.

Reply:
That's because `1.1` had breaking API changes¹ ; so the new constraint should be safe for now, but if we want to be defensive we could add `>= "1.1.0" & < "1.2"`. That could apply to all the other dependencies as well though, the approach of the opam repository maintainers is usually to only add these constraints as they arise.

¹ Actually probably not really, I think they kept compatibility but just with deprecation warnings. Does the Nix build by any chance have warnings-as-errors :scream: ?

Reply:
In my opinion no need to add `>= "1.1.0" & < "1.2"`, it's a problem of the Nix build that ought to be fixed. I think we have warning-as-errors everywhere in this project, including in non-nix builds with `dune build`.

Reply:
> I think we have warning-as-errors everywhere in this project

this is fine for development (and obvious for CI) but shouldn't be enabled for releases and packages (esp. for the "deprecation" warning, we were bitten quite a lot by packages breaking this way in the early days of the opam repo!)

dune has reasonable defaults for the `--profile=dev` and `--profile=release` options though so that should be ok.